apps/src/lib/kits/maker/connectToMakerBoard.js

```diff
@@ -1,16 +1,34 @@
 /**
  * @file Serialport scanning and board controller creation for Maker Toolkit.
  */
+import {getStore} from '../../../redux';
 import {findPortWithViableDevice} from './portScanning';
 import CircuitPlaygroundBoard from './CircuitPlaygroundBoard';
+import {
+  startConnecting,
+  reportConnected,
+  reportConnectionError,
+} from './redux';
 
 /**
  * Establish a connection to a maker board and return a controller for it.
  * @returns {Promise.<CircuitPlaygroundBoard>}}
  */
 export default function connectToMakerBoard() {
+  const store = getStore();
+  const dispatch = store.dispatch.bind(store);
+  let board;
   return Promise.resolve()
+      .then(() => dispatch(startConnecting()))
       .then(findPortWithViableDevice)
-      .then(port => new CircuitPlaygroundBoard(port))
-      .then(board => board.connect().then(() => board));
+      .then(foundPort => new CircuitPlaygroundBoard(foundPort))
+      .then(newBoard => board = newBoard)
+      .then(() => board.connect())
+      .then(() => dispatch(reportConnected()))
+      .then(() => board)
```

Comment:
Okay, I've got a really nice cleanup of this, but it won't be ready for followup review until tomorrow.  Thanks for all the help sorting this out though.